esp/logging/logginglib/logthread.cpp

```diff
@@ -429,7 +429,7 @@ void CLogThread::checkPendingLogs(bool bOneRecOnly)
         }
         //if everything went ok then we should be able to rollover the old logs.
         if (!queueLogError && !bOneRecOnly)
```

Comment:
queueLogError is set if enqueue fails for any one transaction. If enqueue successfully submits pending transaction for processing and fails on another, the files will not be rolled over. The problem is that the successfully submitted transaction, if it does not acknowledged, will be recorded in two tank files with two different GUIDs. When the process restarts, it will see the same transaction multiple times without recognizing that they are the same, and duplicate entries could be submitted.

Should enqueue be changed (or a new version defined) that accepts the original GUID, which is available?

Reply:
This is a good point. Not sure why two different GUIDs are used. We may have to figure it out.